torch/csrc/utils/python_strings.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <Python.h>
+#include <stdexcept>
+#include "object_ptr.h"
+
+// Utilities for handling Python strings. Note that PyString, when defined, is
+// the same as PyBytes.
+
+// Returns true if obj is a bytes/str or unicode object
+// In existing CPython implementations, this does not require the GIL
```

Comment:
We don't depend on this here, but we generally assume that the PyXXX_Check functions don't require the GIL.

However, we depend on THPUtils_checkLong,  THPUtils_unpackLong, and a number of other THPUtils_unpackXXX functions not requiring the GIL, since they're used in TensorMethods.cpp after the GIL is released.